sup3r/bias/bias_transforms.py

```diff
@@ -520,3 +525,247 @@ def local_qdm_bc(data: np.array,
+        Must have datasets "bias_{feature_name}_params",
+        "bias_fut_{feature_name}_params", and "base_{base_dset}_params" that
+        are the parameters to define the statistical distributions to be used
+        to correct the given `data`.
+    lr_padded_slice : tuple | None
+        Tuple of length four that slices (spatial_1, spatial_2, temporal,
+        features) where each tuple entry is a slice object for that axes.
+        Note that if this method is called as part of a sup3r forward pass, the
+        lr_padded_slice will be included automatically in the kwargs for the
+        active chunk. If this is None, no slicing will be done and the full
+        bias correction source shape will be used.
+    no_trend: bool, default=False
+        An option to ignore the trend component of the correction, thus
+        resulting in an ordinary Quantile Mapping, i.e. corrects the bias by
+        comparing the distributions of the biased dataset with a reference
+        datasets. See
+        ``params_mf`` of :class:`rex.utilities.bc_utils.QuantileDeltaMapping`.
+        Note that this assumes that params_mh is the data distribution
+        representative for the target data.
+    """
+    assert data.ndim, "data was expected to be a 3D array"
+    assert data.shape[-1] == time_index.size, \
+        "The last dimension of data should be time"
+
+    params, cfg = get_spatial_bc_presrat(lat_lon,
+                                         base_dset,
+                                         feature_name,
+                                         bias_fp,
+                                         threshold)
+    base = params["base"]
+    bias = params["bias"]
+    bias_fut = params["bias_fut"]
+
+    if lr_padded_slice is not None:
+        spatial_slice = (lr_padded_slice[0], lr_padded_slice[1])
+        base = base[spatial_slice]
+        bias = bias[spatial_slice]
+        bias_fut = bias_fut[spatial_slice]
+
+    if no_trend:
+        mf = None
+    else:
+        mf = bias_fut.reshape(-1, bias_fut.shape[-1])
+    # The distributions are 3D (space, space, N-params)
+    # Collapse 3D (space, space, N) into 2D (space**2, N)
+    QDM = QuantileDeltaMapping(base.reshape(-1, base.shape[-1]),
+                               bias.reshape(-1, bias.shape[-1]),
+                               mf,
+                               dist=cfg['dist'],
+                               relative=relative,
+                               sampling=cfg["sampling"],
+                               log_base=cfg["log_base"])
+
+    # input 3D shape (spatial, spatial, temporal)
+    # QDM expects input arr with shape (time, space)
+    tmp = data.reshape(-1, data.shape[-1]).T
+    # Apply QDM correction
+    tmp = QDM(tmp)
+    # Reorgnize array back from  (time, space) to (spatial, spatial, temporal)
+    tmp = tmp.T.reshape(data.shape)
+
+    tmp = apply_zero_precipitation_rate(tmp, params["base_zero_rate"])
```

Comment:
If we're running GCM future years, shouldn't the zero rate be set using the zero rate from the GCM data? I think the code as written here would fix the zero rate using the historical values. 

Based on this quote from the Pierce paper:
"The GCM predicted future fraction of zero-precipitation days Zgf is calculated using t with the GCM’s original (not bias corrected) future time series."

Reply:
I think we cleared this one already? If you agree, please close (resolve) this conversation.